### ISSUE TITLE
Add spin-up capabilities for `ClimaAtmos` simulations

### ISSUE THREAD

**Issue by sriharshakandala** [CliMA/ClimaAtmos.jl#3327]
For many climate simulations, it is necessary to run a simulation for multiple turn-over times before a statistically stationary state is reached and a usable start-up condition is achieved. For high resolution simulations, this start-up phase can be expensive. It is generally desirable to start the spin-up simulation at a lower resolution and gradually increase the resolution to the target resolution.

**Comment by tapios** [comment]
Yes, this will be a useful capability to have. Before you get started on it, can you please estimate how long it'll take to build this?

**Comment by szy21** [comment]
Would it be easier if we add the capability for reading initial condition from a netcdf file and interpolate? If we do this we won't be constrained by the resolution (both horizontal and vertical).

**Comment by Sbozzolo** [comment]
> Would it be easier if we add the capability for reading initial condition from a netcdf file and interpolate? If we do this we won't be constrained by the resolution (both horizontal and vertical).

ClimaUtilities provides capabilities to accomplish this non-conservatively (using InterpolationsRegridder--multi-linear interpolation), or conservatively with the same vertical levels as the file (using TempestRegridder).

**Comment by charleskawczynski** [comment]
I think interpolating the state / cache is fine, but I think we should still have a way to compute the local geometry given the types of previous data with the new resolution. This way, we don't have errors in mass/energy transport on restarted simulations.

**Comment by sriharshakandala** [comment]
> Would it be easier if we add the capability for reading initial condition from a netcdf file and interpolate? If we do this we won't be constrained by the resolution (both horizontal and vertical).

The plan is to write the restart data to the new (finer) grid.
